e2e/tests/interchain_accounts/base_test.go

```diff
@@ -406,3 +407,129 @@ func (s *InterchainAccountsTestSuite) TestMsgSendTx_SuccessfulTransfer_AfterReop
+	// 	// fund the host account account so it has some $$ to send
+	// 	err := chainB.SendFunds(ctx, interchaintest.FaucetAccountKeyName, ibc.WalletAmount{
+	// 		Address: hostAccount,
+	// 		Amount:  sdkmath.NewInt(testvalues.StartingTokenAmount),
+	// 		Denom:   chainB.Config().Denom,
+	// 	})
+	// 	s.Require().NoError(err)
+	// })
+
+	// t.Run("broadcast MsgSendTx", func(t *testing.T) {
+	// 	// assemble bank transfer message from host account to user account on host chain
+	// 	msgSend := &banktypes.MsgSend{
+	// 		FromAddress: hostAccount,
+	// 		ToAddress:   chainBAccount.FormattedAddress(),
+	// 		Amount:      sdk.NewCoins(testvalues.DefaultTransferAmount(chainB.Config().Denom)),
+	// 	}
+
+	// 	cdc := testsuite.Codec()
+
+	// 	bz, err := icatypes.SerializeCosmosTx(cdc, []proto.Message{msgSend}, icatypes.EncodingProtobuf)
+	// 	s.Require().NoError(err)
+
+	// 	packetData := icatypes.InterchainAccountPacketData{
+	// 		Type: icatypes.EXECUTE_TX,
+	// 		Data: bz,
+	// 		Memo: "e2e",
+	// 	}
+
+	// 	msgSendTx := controllertypes.NewMsgSendTx(controllerAddress, ibctesting.FirstConnectionID, uint64(5*time.Minute), packetData)
+
+	// 	resp := s.BroadcastMessages(
+	// 		ctx,
+	// 		chainA,
+	// 		controllerAccount,
+	// 		msgSendTx,
+	// 	)
+
+	// 	s.AssertTxSuccess(resp)
+
+	// 	// time for the packet to be relayed
+	// 	s.Require().NoError(test.WaitForBlocks(ctx, 5, chainA, chainB))
+	// })
+
+	// t.Run("verify tokens transferred", func(t *testing.T) {
+	// 	balance, err := s.QueryBalance(ctx, chainB, chainBAccount.FormattedAddress(), chainB.Config().Denom)
+	// 	s.Require().NoError(err)
+
+	// 	expected := testvalues.IBCTransferAmount + testvalues.StartingTokenAmount
+	// 	s.Require().Equal(expected, balance.Int64())
+	// })
+
+	channel, err := s.QueryChannel(ctx, chainA, portID, initialChannelID)
+	s.Require().NoError(err)
+
+	// upgrade the channel ordering to UNORDERED
+	upgradeFields := channeltypes.NewUpgradeFields(channeltypes.UNORDERED, channel.ConnectionHops, channel.Version)
+
+	t.Run("execute gov proposal to initiate channel upgrade", func(t *testing.T) {
+		govModuleAddress, err := s.QueryModuleAccountAddress(ctx, govtypes.ModuleName, chainA)
+		s.Require().NoError(err)
+		s.Require().NotNil(govModuleAddress)
+
+		msg := channeltypes.NewMsgChannelUpgradeInit(portID, initialChannelID, upgradeFields, govModuleAddress.String())
+		s.ExecuteAndPassGovV1Proposal(ctx, msg, chainA, controllerAccount)
+	})
+
+	t.Run("verify channel A upgraded and is now unordered", func(t *testing.T) {
+		var channel channeltypes.Channel
+		waitErr := test.WaitForCondition(time.Minute*2, time.Second*5, func() (bool, error) {
+			channel, err = s.QueryChannel(ctx, chainA, portID, initialChannelID)
+			if err != nil {
+				return false, err
+			}
+			return channel.Ordering == channeltypes.UNORDERED, nil
+		})
+		s.Require().NoErrorf(waitErr, "channel was not upgraded: expected %s got %s", channeltypes.UNORDERED, channel.Ordering)
```

Comment:
is waiting for a while not possible here? I.e similar to Carlos's PR https://github.com/cosmos/ibc-go/blob/30b9c38a0a490973547ab6f73a2b0fd9cccc0fc0/e2e/tests/interchain_accounts/upgrades_test.go#L98

Reply:
this was part of the debugging efforts actually, so this kind of polling made more sense. basically the situation was that if we tried to do a transfer before upgrading the ICA channel, the test would fail at hanging on `ChanUpgradeAck`, thus the polling. I'd also be in favour of leaving this in for now until we get a relayer image that is verified to not have a regression.

Reply:
sgtm as long as its tracked!